src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/GpNameAndAddress.java

```diff
@@ -0,0 +1,59 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Example NAD+GP+2750922,295:900'
+ */
+@Getter @Setter @RequiredArgsConstructor
+public class GpNameAndAddress extends Segment{
+
+    public static final String KEY = "NAD";
+    public static final String QUALIFIER = "GP";
+    public static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private @NonNull String identifier;
+    private @NonNull String code;
+
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
```

Comment:
I think this should be `protected` and we should always check the `toEdifact()` from base class

Reply:
I need some more context for that